src/Expander.hs

```diff
@@ -1258,8 +1251,7 @@ runTask (tid, localData, task) = withLocal localData $ do
         Just ready -> do
           for_ ready \(cn, ctor, ts) ->
             addConstructor cn dt ctor ts
-          p <- currentPhase
-          nowValidAt pdest (SpecificPhase p)
+          linkDeclValidity vp scs
```

Comment:
the current phase is now added to the input `ScopeSet`, when `forkEstablishConstructors` is called.